perf/stackdriver.go

```diff
@@ -0,0 +1,177 @@
+package perf
+
+import (
+	"context"
+	"os"
+	"path"
+	"time"
+
+	"cloud.google.com/go/compute/metadata"
+	"contrib.go.opencensus.io/exporter/stackdriver"
+	"github.com/edaniels/golog"
+	"go.opencensus.io/stats/view"
+	"go.opencensus.io/trace"
+
+	"go.viam.com/utils"
+)
+
+const (
+	envVarStackDriverProjectID = "STACKDRIVER_PROJECT_ID"
+)
+
+// Exporter wrapper around Trace and Metric exporter for OpenCensus.
+type Exporter interface {
+	// Start will start the exporting of metrics and return any errors if failed to start.
+	Start() error
+	// Stop will stop all exporting and flush remaining metrics.
+	Stop()
+}
+
+// NewCloudExporter creates a new Stackdriver (Cloud Monitoring) OpenCensus exporter with all options setup.
+func NewCloudExporter(ctx context.Context, logger golog.Logger) (Exporter, error) {
+	sdOpts := stackdriver.Options{
+		Context: ctx,
+		OnError: func(err error) {
+			logger.Errorw("opencensus stackdriver error", "error", err)
+		},
+		// ReportingInterval sets the frequency of reporting metrics to stackdriver backend.
+		ReportingInterval: 60 * time.Second,
+		MetricPrefix:      path.Join("app.viam.com", "opencensus"),
```

Comment:
this is a public library so we can't use this value here. this should be usable by anyone with minimal work as a utility library on top of rpc.

Reply:
Ah good point. There's now a options struct passed in with it and the logger/context.